onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -938,9 +982,60 @@ class UnaryOpSupportChecker : public BaseOpSupportChecker {
           "Sin",
           "Sqrt",
           "Tanh",
+          "QLinearSigmoid",
       });
 }
 
+bool UnaryOpSupportChecker::IsOpSupportedImpl(const InitializedTensorSet& initializers, const Node& node,
+                                              const OpSupportCheckParams& params) const {
+  const auto& op_type = node.OpType();
+  if (op_type != "QLinearSigmoid")
+    return true;
+
```

Comment:
A comment here saying why we exit for all other ops types would be helpful. Also consider splitting out the QLinearSigmoid handling into an op spec private method.